File: VIS_Application/app/Model/UserManagement/AttendanceEntry.ts
export interface IAttendanceEntry {
    Id: number;
    Employee_Id: number;
    Entry_Type: number;
    Remarks: string;
    Entry_Time: Date;
    Date: string;
    actualEntryTime: string;
    grace: number;
    macID: string;
    forWhichDate: Date;
    ipAddress: string;
    source: string;
    isApproved: boolean;
    groupId: number;
    attendancePolicy: string;
    Grade: number;
    ImportRemarks: string;
    Days: string;
    forWhichEntryType: number;
}

export interface IAttendanceEntryGet {
    Id: number;
    Employee_Name: string;
    Password: string;
    Email: string;
    In_Mon_Fri: string;
    Out_Mon_Fri: string;
    In_Sat: string;
    Out_Sat: string;
    Grace: number;
    Active: boolean;
    userType: string;
    joiningDate: Date;
    relevingDate: Date;
    totalCL: number;
    totalSL: number;
    additionalRights: number;
    validForLogin: boolean;
    leaveApproveBy: number;
    employeeCode: string;
    probationTill: Date;
    ActivityId: number;
    CompanyId: number;
    Gender: boolean;
    IsConfirmed: boolean;
    JoiningSalary: number;
    PhotographFileName: string;
    SalaryAfterRevision: number;
    IsSLASigned: boolean;
    ConfirmationDate: Date;
    AppraisalDate: Date;
    IsAppraisalRequired: boolean;
    SLAYear: number;
    FirstName: string;
    MiddleName: string;
    LastName: string;
    JoiningDesignation: number;
    Technology: string;
    IsPasswordChange: boolean;
    OtherTechnology: string;
    SalaryRangeId: number;
    LineManagerId: number;
    TotalCOff: number;
    enrollnumber: number;
    AllowScreenCapture: boolean;
    ActiveDashboard: boolean;
    IsSwitchUser: boolean;
    WorkingLocation: number;
    WorkingLocationAddress: string;
    Education: number;
    CommunicationID: string;
    SatGrace: number;
    IsWebAccess: boolean;
    OtherRemark: string;
    WorksheetThruWeb: boolean;
    IsWorksheetFill: boolean;
    CL: number;
    SL: number;
    IsjoiningLeave: boolean;
    IsConfirmationLeave: boolean;
    EmployeeGradeID: number;
    IsResigned: boolean;
    isMailAlert: boolean;
    IsAllowMouseMovement: boolean;
    accessCardId: number;
    CreatedBy: number;
    CreatedDateTime: Date;
    UpdatedBy: number;
    UpdatedDateTime: Date;
    IsMouseTracking: boolean;
    IsScreenCaptureRemarks: boolean;
    IsHost: boolean;
    IsProductivityTracker: boolean;
    WorkFromHome: boolean;
    attendancePolicy: string;
    IsHostForEmpMaster: boolean;
    ResignedDate: Date;
    CreatedOn: Date;
    UpdatedOn: Date;
    IsActive: boolean;
    
}

export interface IGetTimeDetails {
    TotalOfficeTime: Date;
    TotalPunchInHours: Date;
    TotalBreakTime: Date;
    TotalWorkingTime: Date;
    PunchIn: number;
    InBreak: number;
}

export interface IPunchOutOnNextDay {
    Id: number;
    Employee_Id: number;
    FromDt: Date;
    ToDt: Date;
    Active: Boolean;
    OpenedBy: number;
}

export interface IGetEmployeeName {
    Employee_Id: number;
    Employee_Name: string;
}

File: VIS_Domain/UserManagement/AttendanceEntry.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using VIS_Domain;

namespace VIS_Domain.UserManagement
{
    public class AttendanceEntry : VISBaseEntity
    {
        //Attendance Entry

        public int Employee_Id { get; set; }
        public int Entry_Type { get; set; }
        public string Remarks { get; set; }
        public DateTime Entry_Time { get; set; }
        public string Date { get; set; }
        public string actualEntryTime { get; set; }
        public int grace { get; set; }
        public string macID { get; set; }
        public DateTime forWhichDate { get; set; }
        public string ipAddress { get; set; }
        public string source { get; set; }
        public Boolean isApproved { get; set; }
        public Int64 groupId { get; set; }
        public string attendancePolicy { get; set; }
        public int Grade { get; set; }
        public string ImportRemarks { get; set; }
        public string Days { get; set; }
        public int forWhichEntryType { get; set; }


        //AttendanceEntry Get

        public string Employee_Name { get; set; }
        public string Password { get; set; }
        public string Email { get; set; }
        public string In_Mon_Fri { get; set; }
        public string Out_Mon_Fri { get; set; }
        public string In_Sat { get; set; }
        public string Out_Sat { get; set; }
        public int Grace { get; set; }
        public Boolean Active { get; set; }
        public string userType { get; set; }
        public DateTime joiningDate { get; set; }
        public DateTime relevingDate { get; set; }
        public Decimal totalCL { get; set; }
        public Decimal totalSL { get; set; }
        public int additionalRights { get; set; }
        public Boolean validForLogin { get; set; }
        public int leaveApproveBy { get; set; }
        public string employeeCode { get; set; }
        public DateTime probationTill { get; set; }
        public int ActivityId { get; set; }
        
        public Boolean Gender { get; set; }
        public Boolean IsConfirmed { get; set; }
        public int JoiningSalary { get; set; }
        public string PhotographFileName { get; set; }
        public int SalaryAfterRevision { get; set; }
        public Boolean IsSLASigned { get; set; }
        public DateTime ConfirmationDate { get; set; }
        public DateTime AppraisalDate { get; set; }
        public Boolean IsAppraisalRequired { get; set; }
        public int SLAYear { get; set; }
        public string FirstName { get; set; }
        public string MiddleName { get; set; }
        public string LastName { get; set; }
        public int JoiningDesignation { get; set; }
        public string Technology { get; set; }
        public Boolean IsPasswordChange { get; set; }
        public string OtherTechnology { get; set; }
        public int SalaryRangeId { get; set; }

        public int LineManagerId { get; set; }
        public int TotalCOff { get; set; }
        public int enrollnumber { get; set; }
        public Boolean AllowScreenCapture { get; set; }
        public Boolean ActiveDashboard { get; set; }
        public Boolean IsSwitchUser { get; set; }
        public int WorkingLocation { get; set; }
        public string WorkingLocationAddress { get; set; }
        public int Education { get; set; }
        public string CommunicationID { get; set; }
        public int SatGrace { get; set; }
        public Boolean IsWebAccess { get; set; }
        public string OtherRemark { get; set; }
        public Boolean WorksheetThruWeb { get; set; }
        public Boolean IsWorksheetFill { get; set; }
        public float CL { get; set; }
        public float SL { get; set; }
        public Boolean IsjoiningLeave { get; set; }
        public Boolean IsConfirmationLeave { get; set; }
        public int EmployeeGradeID { get; set; }
        public Boolean IsResigned { get; set; }
        public Boolean isMailAlert { get; set; }
        public Boolean IsAllowMouseMovement { get; set; }
        public int accessCardId { get; set; }
        
        public DateTime CreatedDateTime { get; set; }
        
        public DateTime UpdatedDateTime { get; set; }
        public Boolean IsMouseTracking { get; set; }
        public Boolean IsScreenCaptureRemarks { get; set; }
        public Boolean IsHost { get; set; }
        public Boolean IsProductivityTracker { get; set; }
        public Boolean WorkFromHome { get; set; }
        
        public DateTime IsHostForEmpMaster { get; set; }
        public DateTime ResignedDate { get; set; }
    }

    public static class AttendanceEntryConstants 
    {
        /// Database table name constant.
        ///

        public const string const_Table_Currency = "Attendance_Transaction";

        /// Database table field Name Constants.
        /// 

        public const string const_Field_Employee_Id = "Employee_Id";
        public const string const_Field_Entry_Type = "Entry_Type";
        public const string const_Field_Remarks = "Remarks";
        public const string const_Field_Entry_Time = "Entry_Time";
        public const string const_Field_Date = "Date";
        public const string const_Field_actualEntryTime = "actualEntryTime";
        public const string const_Field_grace = "grace";
        public const string const_Field_macID = "macID";
        public const string const_Field_forWhichDate = "forWhichDate";
        public const string const_Field_ipAddress = "ipAddress";
        public const string const_Field_source = "source";
        public const string const_Field_mouseMovement = "mouseMovement";
        public const string const_Field_isApproved = "isApproved";
        public const string const_Field_groupId = "groupId";
        public const string const_Field_attendancePolicy = "attendancePolicy";
        public const string const_Field_Grade = "Grade";
        public const string const_Field_ImportRemarks = "ImportRemarks";
        public const string const_Field_Days = "Days";
        public const string const_Field_forWhichEntryType = "forWhichEntryType";

        public const string const_Field_Mode = "mode";


        /// Database Procedure and fucntion constants.
        /// 

        public const string const_procAttendanceEntry_Add = "Procattendance_transaction";
        public const string const_procAttendanceEntry_Update = "";
        public const string const_procAttendanceEntry_SelectAll = "ProcPunchOutOnNextDays";
        public const string const_procAttendanceEntry_SelectBaseNonBase = "";
        public const string const_procAttendanceEntry_SelectById = "";

        public const string const_procAttendanceEntry_GetTimer = "ProcLevels_GetAttendance";
        //public const string const_procAttendanceEntry_GetTimer = "ProcLevels_GetAttendance_Timer";
        public const string const_procAttendanceEntry_GetAttendanceEntryType = "procAttendanceEntryType";
        public const string const_procAttendanceEntry_GetCommonConfiguration = "procCommonConfiguration_SelectAll";
        public const string const_procAttendanceEntry_GetPointCounts = "ProcLevels_GetProgressiveDetails";
    }
}
